### PR TITLE
Add Gerrit

### PR DESCRIPTION
**Pull request by reda-alaoui**
This change adds [Gerrit](https://www.gerritcodereview.com/) to the list of `out of the box` platforms.

### CODE REVIEW

**Comment by reda-alaoui** [comment]
Oups, link to Gerrit changes (aka pull requests) seems to be failing.

**Comment by ben-gibson** [comment]
Thanks for this, I'm away for the next week, but I'll look to test/merge/deploy it when I'm back.

**Comment by reda-alaoui** [comment]
> Oups, link to Gerrit changes (aka pull requests) seems to be failing.

The issue is related to #103 . IMO, #103 is marked as fixed but is not.

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@
 <!-- Plugin description -->
 
 A Jetbrains plugin providing shortcuts to open or copy a file, directory or commit in `GitHub`, `Bitbucket`, 
-`GitLab`, `Gitee` `Gitea`, `Gogs` or `Azure`. Custom hosts can also be configured using the URL template syntax.
+`GitLab`, `Gitee` `Gitea`, `Gogs`, `Azure` or `Gerrit`. Custom hosts can also be configured using the URL template syntax.
```
Comment (alexandregaldeano) [inline README.md:14]: Oxford comma here?

File: src/main/kotlin/uk/co/ben_gibson/git/link/platform/PlatformRepository.kt [hunk 2]
```diff
@@ -16,7 +16,8 @@ private val EXISTING_PLATFORMS = setOf(
     Gitea(),
     Gogs(),
     Azure(),
-    Chromium()
+    Chromium(),
+    Gerrit()
```
Comment (alexandregaldeano) [inline src/main/kotlin/uk/co/ben_gibson/git/link/platform/PlatformRepository.kt:20]: Trailing comma here

File: src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -7,7 +7,7 @@
     <depends>Git4Idea</depends>
     <description>
         A Jetbrains plugin open or copy git URLs to a remote host. Supports `GitHub`, `Bitbucket`, `GitLab`, `Gitee`, `Gitea`, `Gogs`,
-        and `Azure` out of the box, while additional hosts can be configured using custom URL templates.
+        `Azure` and `Gerrit` out of the box, while additional hosts can be configured using custom URL templates.
```
Comment (alexandregaldeano) [inline src/main/resources/META-INF/plugin.xml:10]: Oxford comma?

File: src/main/kotlin/uk/co/ben_gibson/git/link/url/template/UrlTemplates.kt [hunk 4]
```diff
@@ -59,5 +59,13 @@ data class UrlTemplates(val fileAtBranch: String, val fileAtCommit : String, val
                 "{remote:url}/commit/{commit}"
             )
         }
+
+        fun gerrit(): UrlTemplates {
+            return UrlTemplates(
+                "{remote:url:protocol}://{remote:url:host}/plugins/gitiles/{remote:url:path}/+/refs/heads/{branch}/{file:path}/{file:name}{line-block:start}#{line:start}{line-block:end}",
```
Comment (alexandregaldeano) [inline src/main/kotlin/uk/co/ben_gibson/git/link/url/template/UrlTemplates.kt:65]: Maybe put `{remote:url:protocol}://{remote:url:host}/plugins/gitiles/{remote:url:path}/+` in a `val`?

File: src/test/kotlin/uk/co/ben_gibson/git/link/url/GerritTest.kt [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        private const val BRANCH = "master"
+        private val COMMIT = Commit("b032a0707beac9a2f24b1b7d97ee4f7156de182c")
+        private val FILE = File("Foo.java", false, "src", false)
+        private val LINE_SELECTION = LineSelection(10, 20)
+
+        @JvmStatic
+        fun urlExpectationsProvider(): Stream<Arguments> = Stream.of(
+            Arguments.of(
+                UrlOptionsFileAtBranch(REMOTE_BASE_URL, FILE, BRANCH, LINE_SELECTION),
+                "https://stash.example.com/plugins/gitiles/foo/bar/+/refs/heads/master/src/Foo.java#10"
```
Comment (alexandregaldeano) [inline src/test/kotlin/uk/co/ben_gibson/git/link/url/GerritTest.kt:28]: Maybe put `https://stash.example.com/plugins/gitiles/foo/bar/+` in a `const val`?

File: src/test/kotlin/uk/co/ben_gibson/git/link/url/GerritTest.kt [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                "https://stash.example.com/plugins/gitiles/foo/bar/+/b032a0707beac9a2f24b1b7d97ee4f7156de182c"
+            ),
+            Arguments.of(
+                UrlOptionsFileAtCommit(REMOTE_BASE_URL, FILE, COMMIT),
+                "https://stash.example.com/plugins/gitiles/foo/bar/+/b032a0707beac9a2f24b1b7d97ee4f7156de182c/src/Foo.java"
+            ),
+            Arguments.of(
+                UrlOptionsCommit(REMOTE_BASE_URL, COMMIT),
+                "https://stash.example.com/plugins/gitiles/foo/bar/+/b032a0707beac9a2f24b1b7d97ee4f7156de182c"
+            )
```
Comment (alexandregaldeano) [inline src/test/kotlin/uk/co/ben_gibson/git/link/url/GerritTest.kt:61]: Trailing comma

**Comment by reda-alaoui** [comment]
@alexandregaldeano , I want to stay consistent with existing code style. Changing the code style should be the object of a separate PR.

**Comment by reda-alaoui** [comment]
`Gerrit` does not support pull request workflow style reviews.

Currently, for any platform, `GitLink` tries to find the local commit on the remote branch to choose between commit url template and branch url template. It makes no sense for `Gerrit`, because `Gerrit` changes are not stored on `refs/heads/{branchname}` but another special references.

TLDR: on Gerrit, the commit url template must **always** be used.

Because of that, I added a boolean `pullRequestWorkflowSupported` to `Platform`. True by default. If false, the commit url template will always be used instead of the branch url template.

**Comment by reda-alaoui** [comment]
Thank you !
